trunner/test/harnesses/test_micropython.py

```diff
@@ -0,0 +1,473 @@
+#
+# Phoenix-RTOS test runner
+#
+# Tests for MicroPython Test Suite harness
+#
+# Copyright 2022 Phoenix Systems
+# Authors: Piotr Nieciecki
+#
+
+import os
+import threading
+import pytest
+import re
+import pexpect
+import pexpect.fdpexpect
+import socket
+
+from trunner.harnesses.micropython import MicropythonStandardHarness
+from trunner.harnesses.micropython import MicropythonReplHarness
+from trunner.harnesses.micropython import create_regex
+from trunner.harnesses.micropython import standard_regex_escape
+from trunner.harnesses.common import TestResult
+
+TestResult.__test__ = False
+
+PROMPT = "\r\u001b[0J(psh)% "
+
+MICROPYTHON = '/bin/micropython'
+UPYTHON_HEADER = '''MicroPython 6d2ae58-dirty on 2022-07-27; linux version\r
+Use Ctrl-D to exit, Ctrl-E for paste mode\r
+>>> '''
+UPYTHON_PROMPT = "\r\n>>>"
+EXP_UPYTHON_HEADER = "MicroPython \\.\\+ version\nUse \\.\\+\n>>> "
+
+
+def check_result(result, exp_result):
+    result = result[0]
+
+    assert result.name == exp_result.name
+    assert result.status == exp_result.status
+
+    if exp_result.msg is not None:
+        assert result.msg == exp_result.msg
+
+
+class MicropythonMock:
+
+    SOCKET_FILE = "/tmp/test_micropython"
+
+    def __init__(self):
+        self.success = True
+
+    def send(self, text):
+        self.socket.send(text.encode("ascii"))
+
+    def recv_exp(self, exp_output):
+        res = self.socket.recv(len(exp_output))
+        res = res.decode("ascii")
+        if res != exp_output:
+            self.success = False
+            self.problem_msg = f"Error - callback didn't received \"{exp_output}\", but got \"{res}\""
+
+    def get_output(self, harness, callback, *callback_args):
+        """Function uses callback function to communicate with specified harness. Returns harness result"""
+
+        self.callback = callback
+
+        s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        if os.path.exists(self.SOCKET_FILE):
+            os.remove(self.SOCKET_FILE)
+        s.bind(self.SOCKET_FILE)
+
+        lock = threading.Lock()
+        lock.acquire()
+
+        thread = threading.Thread(target=self.__start_callback, args=(lock, *callback_args))
```

Comment:
pass `callback` directly to the __start_callback, not by setting it as an attribute